### PR TITLE
[Core] Moving some containers code to cpp 

### PR DESCRIPTION
**Pull request by loumalouomega**
This avoid to include kratos_components, reducing a bit the compilation time

### CODE REVIEW

**Comment by philbucher** [comment]
+1, makes sense to me

we should carefully review to not break things

**Comment by loumalouomega** [comment]
> 
> 
> +1, makes sense to me
> 
> we should carefully review to not break things

I forgot the KRATOS_API

**Comment by loumalouomega** [comment]
@tteschemacher can you approve this in order to later merge #6367?

File: kratos/containers/data_value_container.cpp [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 
 // Project includes
-#include "includes/define.h"
+#include "containers/variable_component.h"
```
Comment (tteschemacher) [inline kratos/containers/data_value_container.cpp:22]: One question why did you move this to the cpp?

**Review by tteschemacher — APPROVED** [review]
The changes are obvious.

Thanks @loumalouomega

File: kratos/containers/data_value_container.cpp (hunk 1 above)
Comment (loumalouomega) [reply kratos/containers/data_value_container.cpp:22]: When we had compilation problems due to recursive dependencies
